### PR TITLE
society library

### PR DESCRIPTION
**Pull request by johnwarden**
- **Create script to import society library date map**
- **Improve error reporting in globalbrain.ts**

### CODE REVIEW

File: other/import-society-library-debatemap.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import fs from 'fs'
+import { createPost } from '#app/post.js'
```
Comment (JohannesNakayama) [inline other/import-society-library-debatemap.ts:2]:
```suggestion
import { createPost } from '../app/post.ts'
```

File: other/import-society-library-debatemap.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+import { invariant } from '#app/utils/misc.tsx'
```
Comment (JohannesNakayama) [inline other/import-society-library-debatemap.ts:3]:
```suggestion
import { invariant } from '../app/utils/misc.tsx'
```

File: other/import-society-library-debatemap.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+console.log('debateFilename', debateFilename)
+
+var topLevelQuestion: Question = JSON.parse(
+	fs.readFileSync(debateFilename, 'utf-8'),
+) as Question
+
+console.log('Got debate map for question', topLevelQuestion)
+
+const societyLibraryTag = 'society-library'
+const userId = 'developer'
```
Comment (fdietze) [inline other/import-society-library-debatemap.ts:18]: pass userId as an environment variable

**Review by JohannesNakayama — APPROVED** [review]

File: justfile [hunk 4]
```diff
@@ -57,6 +57,9 @@ reseed:
 import-hn:
 	npx tsx import-hn.ts ./other/hn-data/*.json.gz
 
+import-society-library:
+	cd other && npx tsx import-society-library-debatemap.ts society-library-openai-debatemap.json
```
Comment (fdietze) [inline justfile:61]: don't hardcode the json file here, instead leave it out and pass it as an argument to the just command

**Review by fdietze — APPROVED** [review]

**Pull request merged**